### ISSUE TITLE
Upgrade to Mule 4.6 and Java 17

### ISSUE THREAD

**Issue by adesjardin** [avioconsulting/mule-opentelemetry-module#152]
MuleSoft has released the 4.6 runtime which is the first version to support Java 17.  We should validate against 4.6 using Java 17 and implement an approach for building and releasing a parallel version to support this.

**Comment by Kevin44** [comment]
Any updates or a projected timeline on this?

**Comment by holiday-sunrise** [comment]
any idea when java17 will be supported ?

**Comment by adesjardin** [comment]
@holiday-sunrise - We have it as one of the top issues on our backlog right now.  We're hoping to have a release with support for Java 17 available in the next 2-3 weeks.

**Comment by holiday-sunrise** [comment]
Is there a pre-release so we can help testing

**Comment by aticlo** [comment]
Hey.  Is there a date for this Release supporting JDK 17?

**Comment by diepet** [comment]
Hi all, is there some update about the resolution of this issue?

**Comment by edsonmiguelrs** [comment]
Hi all, is the some update about this issue?

**Comment by diepet** [comment]
Hi does the new 2.3.0 version of this library supporting the JDK17 and Mule 4.6?

**Comment by aticlo** [comment]
The release notes for 2.3.0 don't mention the support for jdk 17 neither runtime 4.6 support.
Also this version is not present in the mvn repo.

**Comment by manikmagar** [comment]
@diepet 2.3.0 does not support Java 17/Runtime 4.6 yet. @aticlo However, this version is available on Maven Central - 
https://repo1.maven.org/maven2/com/avioconsulting/mule/mule-opentelemetry-module/2.3.0/ 

JDK17 support is in the queue to be upgraded very soon. Before we can upgrade this, we need to update and upgrade some dependencies and processes.

**Comment by JYOTHI-SHRAVZ** [comment]
Hi , do you have a date for the Java 17 release?

**Comment by manikmagar** [comment]
@JYOTHI-SHRAVZ We are in the process of releasing 2.4.0 this week. The next is 2.5.0 with Java 17 support. The PR is already out there - #212. It should be sometime next week or sooner. 

Meantime, if you would like to test the module with Java 17, you could follow these steps - 

1. Clone this repo to your local
2. Switch to branch `feat/java-17-support`
3. Build and install the module in local maven repo `./mvnw clean install`
4. Update dependency in a local project to `2.5.0-SNAPSHOT`
5. Test with Java 17 and the appropriate Mule Runtime version

If you do that, we would appreciate any feedback you may have.

**Comment by manikmagar** [comment]
Release Java 17 support with [2.5.0](https://github.com/avioconsulting/mule-opentelemetry-module/releases/tag/v2.5.0).

**Comment by aticlo** [comment]
Good morning. Isn't the version 2.5.0 and 2.5.1 going to be published to https://mvnrepository.com/artifact/com.avioconsulting/mule-opentelemetry-module ?

**Comment by manikmagar** [comment]
> Good morning. Isn't the version 2.5.0 and 2.5.1 going to be published to https://mvnrepository.com/artifact/com.avioconsulting/mule-opentelemetry-module ?

Apologies for missing your previous comment, @aticlo. I hope you have figured out already that you are referring to an older group ID (`com.avioconsulting`). With 2.0, the group ID was changed to `com.avioconsulting.mule`—https://mvnrepository.com/artifact/com.avioconsulting.mule/mule-opentelemetry-module. Feel free to open new issue if you there are still problems. Thanks!
